Change behavior of loading new posts in the reader

nbradbury opened the pull request:
Fix #2133 - rewrote how new posts are loaded and displayed in the reader. When new posts are loaded, the current position is retained and a "New posts" bar appears at the top of the screen which animates out after a few seconds. Tapping this bar scrolls the new posts into view.

Making this work required avoiding the use of `notifyDataChanged()` in the adapter, since that would lose the current position (and was very inefficient). The [new code](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2133-reader-remember-position/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L625) relies on `notifyItemInserted()` and `notifyItemChanged()` instead.

roundhill commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -665,7 +665,7 @@
 
     <!-- TextView labels -->
-    <string name="reader_label_new_posts">Show new posts</string>
+    <string name="reader_label_new_posts">&#8593; New posts</string>
```
The up arrow looks a little weird (thin and too high imo), could we use a `drawableLeft` with a proper icon?

roundhill commented on WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java:
```diff
@@ -175,6 +177,8 @@ public void setArguments(Bundle args) {
                 mTagPreviewHistory.push(getCurrentTagName());
             }
+
+            // TODO: save KEY_RESTORE_POSITION
```
Taking care of this later?

nbradbury commented:
@roundhill Ready for re-review! I redesigned the "new posts" bar to use a drawable as you (correctly) suggested. That "todo" was a misplaced reminder to me to save the current position in onSaveInstanceState which I already added (so I removed the todo).

nbradbury commented:
@roundhill Good call on the material up arrow icons. I just switched over to them.

roundhill commented:
There's been a few times where I've opened the reader or switched tags, when the list was at the top. When I did a PTR, the prompt showed that there were new posts, but tapping it didn't show any new posts. I think the prompt was incorrectly shown.

nbradbury commented:
I think I found the problem: the "new posts" bar would appear if there were *changed* posts even when there weren't any new ones. Fixed in 4a55292.

roundhill commented:
Grand, let's :shipit:
